src/Client.test.js

```
@@ -0,0 +1,149 @@
+import AMQP, * as amqpMocks from './network/AMQP';
+import * as api from './config/api';
+import Client from './Client';
+
+jest.mock('./network/AMQP');
+
+const mockToken = 'my-authorization-token';
+const mockThing = {
+  id: 'abcdef1234568790',
+  name: 'my-device',
+  schema: [{
+    sensorId: 0,
+    valueType: 3,
+    unit: 0,
+    typeId: 65521,
+    name: 'bool-sensor',
+  }],
+};
+
+const errors = {
+  // Client methods
+  register: 'error registering thing',
+  unregister: 'error unregistering thing',
+  // AMQP methods
+  start: 'error while starting AMQP connection',
+  stop: 'error while stopping AMQP connection',
+  publishMessage: 'error publishing message on broker',
+  subscribeTo: 'error subscribing consumer on queue',
+  unsubscribeConsumer: 'error unsubscribing consumer',
+};
+
+const registerUseCases = [
+  {
+    testName: 'thing registered successfully',
+    amqpOptions: {
+      responseMessage: { id: mockThing.id },
+    },
+  },
+  {
+    testName: 'fail to register thing on cloud',
```

In unit tests is common we have names containing:
A precondition (what scenario has to be to you execute the action), 
an action (usually the function that you are calling)
and an assertion (a verification that what you are expecting)

These names are just putting the expected behavior.
This is important because we can know, if this test fail later versions, in what conditions we can reproduce it.

An example:
`When device ID is not valid in a register command should raise an exception of argument invalid`

---

Done! :smile: